🌐 Add Turkish translation (#470)
jowilf/starlette-admin: docs/tutorial/authentication/index.tr.md

# Kimlik Doğrulama ve Yetkilendirme

> Yönetici arayüzünüzü istenmeyen kullanıcılardan korumak için, [AuthProvider][starlette_admin.auth.AuthProvider] sınıfını genişleterek bir `Kimlik Doğrulama Sağlayıcısı` oluşturabilirsiniz ve admin uygulamanızı tanımlarken `auth_provider` özelliğine atayabilirsiniz.

Reviewer: ```suggestion
Yönetici arayüzünüzü istenmeyen kullanıcılardan korumak için [AuthProvider][starlette_admin.auth.AuthProvider] sınıfını genişleterek bir `Kimlik Doğrulama Sağlayıcısı` oluşturabilir ve admin uygulamanızı tanımlarken `auth_provider` özelliğine atayabilirsiniz.
```

## Kullanıcı Adı ve Şifre ile Kimlik Koğrulama

> [AuthProvider][starlette_admin.auth.AuthProvider] varsayılan olarak `username` ve `password` alanları içeren bir giriş formu sağlar. Bu kimlik doğrulama yöntemini tam olarak desteklemek için, özel `Kimlik Doğrulama Sağlayıcı`nızda aşağıdaki yöntemleri uygulamanız gerekir:

Reviewer: ```suggestion
[AuthProvider][starlette_admin.auth.AuthProvider] varsayılan olarak `username` ve `password` alanları içeren basit bir giriş formu sağlar. Bu kimlik doğrulama yöntemini tam olarak desteklemek için, özel `Kimlik Doğrulama Sağlayıcı`nızda aşağıdaki yöntemleri uygulamanız gerekir:
```

* [is_authenticated][starlette_admin.auth.BaseAuthProvider.is_authenticated]: Bu metod, gelen her isteği doğrulamak için çağrılacaktır.

> * [get_admin_user][starlette_admin.auth.BaseAuthProvider.get_admin_user]: Bağlı olan kullanıcı `adı` ve/veya `profil` bilgisini döndürür.

Reviewer: ```suggestion
* [get_admin_user][starlette_admin.auth.BaseAuthProvider.get_admin_user]: Bağlı olan kullanıcı `adı` ve/veya `avatar` bilgisini döndürür.
```

> * [get_admin_config][starlette_admin.auth.BaseAuthProvider.get_admin_config]: Bağlı olan kullanıcıya göre veya başka koşullara göre `logo_url` veya `app_title` bilgisini döndürür.

Reviewer: ```suggestion
* [get_admin_config][starlette_admin.auth.BaseAuthProvider.get_admin_config]: Bağlı olan kullanıcıya veya başka koşullara göre `logo_url` veya `app_title` bilgisini döndürür.
```

        "roles": ["read", "create", "edit", "action_make_published"],
    },
    "viewer": {"name": "Viewer", "avatar": "guest.png", "roles": ["read"]},
}

class UsernameAndPasswordProvider(AuthProvider):
    """
    Bu sadece göstermek için bir örnektir, kullanıcı kimlik
    bilgilerini kaydetmek ve doğrulamak için daha iyi bir yol değildir.
    """

    async def login(
        self,
        username: str,
        password: str,
        remember_me: bool,
        request: Request,
        response: Response,
    ) -> Response:
        if len(username) < 3:
            """Form verisi doğrulama"""
            raise FormValidationError(
                {"username": "Ensure username has at least 03 characters"}
            )

        if username in users and password == "password":
            """`username` bilgisini session'a kaydedin"""
            request.session.update({"username": username})
            return response

        raise LoginFailed("Invalid username or password")

    async def is_authenticated(self, request) -> bool:
        if request.session.get("username", None) in users:
            """
            İsteğin durumunda bulunan `user` nesnesini kaydedin.

>             Daha sonra bağlı kullanicinin erişimin' kısıtlamak için kullanılabilir.

Reviewer: ```suggestion
            Daha sonra bağlı kullanıcının erişimini kısıtlamak için kullanılabilir.
```

            )
        )

    def get_middleware(self, admin: "BaseAdmin") -> Middleware:
        return Middleware(
            AuthMiddleware, provider=self, allow_paths=["/auth0/authorize"]
        )

```

Çalışan bir örnek için [`https://github.com/jowilf/starlette-admin/tree/main/examples/authlib`](https://github.com/jowilf/starlette-admin/tree/main/examples/authlib) sayfasına bakabilirsiniz.

`AuthProvider` aşağıdaki gibi yönetici arayüzünüze eklenebilir:

```python
admin = Admin(
    engine,
    title="Example: Authentication",
    auth_provider=MyAuthProvider(),
    middlewares=[Middleware(SessionMiddleware, secret_key=SECRET)],
)
```

## Authorization - Yetkilendirme

### Tüm Görünümler İçin

Her [görünüm][starlette_admin.views.BaseView] geçerli kullanıcının erişimini kısıtlamak için kullanılabilecek [is_accessible][starlette_admin.views.BaseView.is_accessible] metodunu kullanır.

```python

```python
from starlette_admin import CustomView
from starlette.requests import Request

class ReportView(CustomView):

    def is_accessible(self, request: Request) -> bool:
        return "admin" in request.state.user["roles"]
```

!!! important "Önemli"

>     Görünüm erişilemez olduğunda, menü yapısında yer almaz.

Reviewer: ```suggestion
    Görünüm erişilemez olduğunda menü yapısında yer almaz.
```

### [ModelView][starlette_admin.views.BaseModelView] İçin

> In [ModelView][starlette_admin.views.BaseModelView], içinde, geçerli kullanıcının erişimini kısıtlamak için aşağıdaki yöntemleri geçersiz kılabilirsiniz:

Reviewer: ```suggestion
[ModelView][starlette_admin.views.BaseModelView] içinde, geçerli kullanıcının erişimini kısıtlamak için aşağıdaki yöntemleri geçersiz kılabilirsiniz:
```

* `can_view_details`: Bir öğenin tam ayrıntılarını görüntüleme izni.
* `can_create`: Bir öğeyi oluşturma izni.
* `can_edit`: Bir öğeyi düzenleme izni.
* `can_delete`: Bir öğeyi silme izni.

> * `is_action_allowed`: ismi verilen <abbr title="action">toplu işlem</abbr>in izin verilip verilmediğini doğrulayın.
> * `is_row_action_allowed`: ismi verilen <abbr title="action">satır işlem</abbr>inin izin verilip verilmediğini doğrulayın.

Reviewer: ```suggestion
* `is_action_allowed`: belirli bir <abbr title="action">toplu işlem</abbr>in izin verilip verilmediğini doğrulayın.
* `is_row_action_allowed`: belirli bir <abbr title="action">satır işlem</abbr>inin izin verilip verilmediğini doğrulayın.
```
